publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/batch/dynamodb/jobs/UpdateVerificationStatusJob.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    public UpdateVerificationStatusJob(ResourceService resourceService, CristinClient cristinClient) {
+        this.resourceService = resourceService;
+        this.cristinClient = cristinClient;
+    }
+
+    @Override
+    public void executeBatch(List<BatchWorkItem> workItems) {
+        workItems.stream()
+            .map(this::extractIdentifier)
+            .map(this::fetchResource)
+            .flatMap(this::updateVerificationStatus)
+            .forEach(this::persistResourceUpdate);
+    }
+
+    @Override
+    public String getJobType() {
+        return JOB_TYPE;
+    }
+
+    private SortableIdentifier extractIdentifier(BatchWorkItem workItem) {
+        var sortKey = workItem.dynamoDbKey().sortKey();
+        var parts = sortKey.split(KEY_FIELDS_DELIMITER);
+        return new SortableIdentifier(parts[IDENTIFIER_INDEX]);
+    }
+
+    private Resource fetchResource(SortableIdentifier identifier) {
+        try {
+            return resourceService.getResourceByIdentifier(identifier);
+        } catch (NotFoundException e) {
+            throw new RuntimeException("Resource not found: " + identifier, e);
+        }
+    }
+
+    private Stream<Resource> updateVerificationStatus(Resource resource) {
+        return Optional.ofNullable(resource.getEntityDescription())
+                   .map(EntityDescription::getContributors)
+                   .map(this::updateContributors)
+                   .filter(updatedContributors -> hasChanges(resource.getEntityDescription().getContributors(),
+                                                             updatedContributors))
+                   .map(updatedContributors -> applyContributorUpdates(resource, updatedContributors))
+                   .stream();
+    }
+
+    private List<Contributor> updateContributors(Collection<Contributor> contributors) {
+        return contributors.stream()
+                   .map(this::updateContributorVerificationStatus)
+                   .toList();
+    }
+
+    private Resource applyContributorUpdates(Resource resource, List<Contributor> updatedContributors) {
+        resource.getEntityDescription().setContributors(updatedContributors);
+        return resource;
+    }
+
+    private void persistResourceUpdate(Resource resource) {
+        resourceService.updateResource(resource, UserInstance.fromPublication(resource.toPublication()));
+        logger.info("Updated verification status for resource: {}", resource.getIdentifier());
+    }
+
+    private boolean hasChanges(Collection<Contributor> original, Collection<Contributor> updated) {
+        return !original.equals(updated);
+    }
+
+    private Contributor updateContributorVerificationStatus(Contributor contributor) {
+        return extractCristinId(contributor)
+                   .map(this::fetchVerificationStatus)
+                   .flatMap(status -> applyVerificationStatus(contributor, status))
+                   .orElse(contributor);
+    }
```

Comment:
When we change the verification status, should we log that? I.e. log both the Cristin ID and the old/new values.